src/routes/auth/callback/+server.ts

```diff
@@ -0,0 +1,80 @@
+import { OAuth2RequestError } from 'arctic';
+import { generateId } from 'lucia';
+import { eq } from 'drizzle-orm';
+
+import { lucia, googleAuth } from '$lib/server/auth';
+import { db } from '$lib/db/db.server';
+import { userTable } from '$lib/db/schema';
+
+import type { RequestEvent } from '@sveltejs/kit';
+
+export async function GET(event: RequestEvent): Promise<Response> {
+	const code = event.url.searchParams.get('code');
+	const state = event.url.searchParams.get('state');
+	const storedState = event.cookies.get('google_state');
+	const storedCodeVerifier = event.cookies.get('code_verifier');
+
+	if (!code || !state || !storedState || state !== storedState || !storedCodeVerifier) {
+		return new Response('User Cancelled', {
+			status: 302,
+			headers: {
+				location: '/'
+			}
+		});
+	}
+
+	try {
+		const tokens = await googleAuth.validateAuthorizationCode(code, storedCodeVerifier);
+		const response = await fetch('https://openidconnect.googleapis.com/v1/userinfo', {
+			headers: {
+				Authorization: `Bearer ${tokens.accessToken}`
+			}
+		});
+		const googleUser = await response.json();
```

Comment:
```suggestion
		const { name, email, sub: googleId } = await response.json();
```

make sure to change the parts this is used too